sql/src/test/java/io/crate/executor/transport/TransportShardUpsertActionTest.java

```diff
@@ -226,10 +227,33 @@ public void testProcessGeneratedColumnsWithInvalidValueNoValidation() throws Exc
                 .put("day", 1448274317000L)
                 .map();
 
+
         transportShardUpsertAction.processGeneratedColumns(generatedColumnTableInfo, updatedColumns, updatedGeneratedColumns, false);
     }
 
     @Test
+    public void testProcessValidateArrayGeneratedColumns() throws Exception {
+        // test no exception are thrown when validating array generated columns
+        Map<String, Object> updatedColumns = MapBuilder.<String, Object>newMapBuilder()
+            .put("obj", MapBuilder.<String, Object>newMapBuilder().put("arr", new Object[]{1}).map())
+            .map();
+
+        Map<String, Object> updatedGeneratedColumns = MapBuilder.<String, Object>newMapBuilder()
+            .put("arr", new Object[]{1})
+            .map();
+
+        DocTableInfo docTableInfo = new TestingTableInfo.Builder(
+            new TableIdent(null, "generated_column"),
+            new Routing(Collections.EMPTY_MAP))
```

Comment:
please use the `emptyMap()` method instead of the not-type-safe EMPTY_MAP constant